core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -115,6 +120,53 @@ public static Schema schema(StructType unifiedPartitionType) {
         LAST_UPDATED_SNAPSHOT_ID);
   }
 
+  /**
+   * Generates the partition stats file schema based on a combined partition type which considers
+   * all specs in a table. (For format version 3 and above)
+   *
+   * @param unifiedPartitionType unified partition schema type. Could be calculated by {@link
+   *     Partitioning#partitionType(Table)}.
+   * @return a schema that corresponds to the provided unified partition type.
+   */
+  public static Schema schemaV3Plus(StructType unifiedPartitionType) {
+    Preconditions.checkState(!unifiedPartitionType.fields().isEmpty(), "Table must be partitioned");
+    return new Schema(
+        NestedField.required(PARTITION_FIELD_ID, PARTITION_FIELD_NAME, unifiedPartitionType),
+        SPEC_ID,
+        DATA_RECORD_COUNT,
+        DATA_FILE_COUNT,
+        TOTAL_DATA_FILE_SIZE_IN_BYTES,
+        NestedField.required(
```

Comment:
As per the spec PR (these fields are required now)
https://github.com/apache/iceberg/pull/12098